src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -51,27 +56,75 @@
+     * @param modid the mod id calling context
+     * @return a {@link RegistryObject} that stores the value of an object from a registry once it is ready
+     * @see #of(ResourceLocation, ResourceKey, String)
+     * @see #of(ResourceLocation, IForgeRegistry)
+     * @see #of(ResourceLocation, ResourceLocation, String)
+     */
+    public static <T, U extends T> RegistryObject<U> optional(final ResourceLocation name, final ResourceKey<? extends Registry<T>> registryKey, String modid) {
+        return new RegistryObject<>(name, registryKey.location(), modid, true);
+    }
+
+    /**
+     * Factory for a {@link RegistryObject} that stores the value of an object from a registry once it is ready.
+     * <p>
+     * If a registry with the given name cannot be found, an exception will be thrown when trying to fill this RegistryObject.
+     * Use {@link #optional(ResourceLocation, ResourceLocation, String)} for RegistryObjects of optional registries.
+     *
+     * @param name the name of the object to look up in a registry
+     * @param registryName the name of the registry. Supports lookups on {@link BuiltinRegistries}, {@link Registry}, and {@link RegistryManager#ACTIVE}.
+     * @param modid the mod id calling context
+     * @return a {@link RegistryObject} that stores the value of an object from a registry once it is ready
+     * @see #optional(ResourceLocation, ResourceLocation, String)
+     * @see #of(ResourceLocation, IForgeRegistry)
+     * @see #of(ResourceLocation, ResourceKey, String)
      */
     public static <T, U extends T> RegistryObject<U> of(final ResourceLocation name, final ResourceLocation registryName, String modid) {
-        return new RegistryObject<>(name, registryName, modid);
+        return new RegistryObject<>(name, registryName, modid, false);
+    }
+
+    /**
+     * Factory for a {@link RegistryObject} that stores the value of an object from a registry once it is ready.
+     * <p>
+     * If a registry with the given name cannot be found, it will be silently ignored and this RegistryObject will not be filled.
+     * Use {@link #of(ResourceLocation, ResourceLocation, String)} for RegistryObjects that should throw exceptions on missing registry.
+     *
+     * @param name the name of the object to look up in a registry
+     * @param registryName the name of the registry. Supports lookups on {@link BuiltinRegistries}, {@link Registry}, and {@link RegistryManager#ACTIVE}.
+     * @param modid the mod id calling context
+     * @return a {@link RegistryObject} that stores the value of an object from a registry once it is ready
+     * @see #of(ResourceLocation, ResourceLocation, String)
+     * @see #of(ResourceLocation, IForgeRegistry)
+     * @see #of(ResourceLocation, ResourceKey, String)
+     */
+    public static <T, U extends T> RegistryObject<U> optional(final ResourceLocation name, final ResourceLocation registryName, String modid) {
```

Comment:
Same comment for `ofOptional`. Formatting: braces on newlines.
```suggestion
    public static <T, U extends T> RegistryObject<U> optional(final ResourceLocation name, final ResourceLocation registryName, String modid)
    {
```